cve_bin_tool/checkers/binutils.py

```diff
@@ -0,0 +1,132 @@
+        # dlltool - Creates files for building and using DLLs.
+        # gold - A new, faster, ELF only linker, still in beta test.
+        r"restart link with --incremental-full",
+        # gprof - Displays profiling information.
+        r"%s: gmon.out file is missing histogram",
+        # nlmconv - Converts object code into an NLM.
+        # nm - Lists symbols from object files.
+        r"Using the --size-sort and --undefined-only options together",
+        # objcopy - Copies and translates object files.
+        r"alloc, load, noload, readonly, debug, code, data, rom, share, contents, merge, strings",
+        # objdump - Displays information from object files.
+        # ranlib - Generates an index to the contents of an archive.
+        # readelf - Displays information from any ELF format object file.
+        # size - Lists the section sizes of an object or archive file.
+        # strings - Lists printable strings from files.
+        r"can't set BFD default target to `%s': %s"
+        # strip - Discards symbols.
+        # windmc - A Windows compatible message compiler.
+        # windres - A compiler for Windows resource files.
+    ]
+
+    for line in lines:
+        for signature in signatures:
+            pattern = re.compile(signature)
+            if pattern.search(line):
+                return 1
+    return 0
+
+
+def get_version(lines, filename):
+    """returns version information for binutiles as found in a given file.
+
+    VPkg: gnu, binutils
+    """
+    version_info = dict()
+
+    util_names = [
+        # command line utils
+        "ld",  # the GNU linker.
+        "as",  # the GNU assembler.
+        "addr2line",  # Converts addresses into filenames and line numbers.
+        "ar",  # A utility for creating, modifying and extracting from archives.
+        "c++filt",  # Filter to demangle encoded C++ symbols.
+        "dlltool",  # Creates files for building and using DLLs.
+        "gold",  # A new, faster, ELF only linker, still in beta test.
+        "gprof",  # Displays profiling information.
+        "nlmconv",  # Converts object code into an NLM.
+        "nm",  # Lists symbols from object files.
+        "objcopy",  # Copies and translates object files.
+        "objdump",  # Displays information from object files.
+        "ranlib",  # Generates an index to the contents of an archive.
+        "readelf",  # Displays information from any ELF format object file.
+        "size",  # Lists the section sizes of an object or archive file.
+        "strings",  # Lists printable strings from files.
+        "strip",  # Discards symbols.
+        "windmc",  # A Windows compatible message compiler.
+        "windres",  # A compiler for Windows resource files.
+        # distro-specific names
+        "ld.bfd",  # as seen on ubuntu
+    ]
+
+    # check if it's the library instead of the command line utils
+    lib_pattern = re.compile(r"libbfd-((\d+\.)*\d+)[\d\w.-]*so")
```

Comment:
Clever!  I'm guessing from the fact that you've got a working centos test, this resolves the issue I had in #54 where it only worked for Ubuntu.